src/activity_actor.h

```diff
@@ -67,6 +75,102 @@ class activity_actor
         virtual void serialize( JsonOut &jsout ) const = 0;
 };
 
+class dig_activity_actor : public activity_actor
+{
+    private:
+        int moves_total;
+        /** location of the dig **/
+        tripoint location;
+        std::string result_terrain;
+        tripoint byproducts_location;
+        int byproducts_count;
+        std::string byproducts_item_group;
+
+    public:
+        dig_activity_actor(
+            int dig_moves, const tripoint &dig_loc,
+            const std::string &resulting_ter, const tripoint &dump_loc,
+            int dump_spawn_count, const std::string &dump_item_group
+        ):
+            moves_total( dig_moves ), location( dig_loc ),
+            result_terrain( resulting_ter ),
+            byproducts_location( dump_loc ),
+            byproducts_count( dump_spawn_count ),
+            byproducts_item_group( dump_item_group ) {}
+
+        activity_id get_type() const override {
+            return activity_id( "ACT_DIG" );
+        }
+
+        void start( player_activity &act, Character & ) override;
+        void do_turn( player_activity &, Character & ) override;
+        void finish( player_activity &act, Character &who ) override;
+        bool can_resume_with( const activity_actor &other, const Character & ) const override;
+
+        bool operator==( const dig_activity_actor &other ) const {
```

Comment:
I'd suggest making this a normal function instead of `operator==`, otherwise people may get surprised when they find out `moves_total` is not compared when they compare two activity actors in other scenarios.

Reply:
Thanks, I agree with you, I will do this.